lib/errors/error-collector.js

```diff
@@ -124,48 +190,30 @@ class ErrorCollector {
       return
     }
 
-    // TODO: Prob shouldn't do any work if errors fully disabled.
-
     // collect user errors even if status code is ignored
     let collectedErrors = 0
     let expectedErrors = 0
 
     // errors from noticeError are currently exempt from
     // ignore and exclude rules
-    if (transaction.userErrors.length) {
-      for (let i = 0; i < transaction.userErrors.length; i++) {
-        const exception = transaction.userErrors[i]
-        if (this.collect(transaction, exception)) {
-          ++collectedErrors
-        }
-      }
-    }
+    collectedErrors = this._processUserErrors(transaction, collectedErrors)
 
     const isErroredTransaction = urltils.isError(this.config, transaction.statusCode)
     const isIgnoredErrorStatusCode = urltils.isIgnoredError(this.config, transaction.statusCode)
 
-    const isExpectedErrorStatusCode = urltils.isExpectedError(this.config, transaction.statusCode)
-
     // collect other exceptions only if status code is not ignored
     if (transaction.exceptions.length && !isIgnoredErrorStatusCode) {
-      for (let i = 0; i < transaction.exceptions.length; i++) {
-        const exception = transaction.exceptions[i]
-        if (this.collect(transaction, exception)) {
-          ++collectedErrors
-          // if we could collect it, then check if expected
-          if (
-            isExpectedErrorStatusCode ||
-            errorHelper.isExpectedException(transaction, exception.error, this.config, urltils)
-          ) {
-            ++expectedErrors
-          }
-        }
-      }
-    } else if (isErroredTransaction && this.collect(transaction)) {
-      ++collectedErrors
-      if (isExpectedErrorStatusCode) {
-        ++expectedErrors
-      }
+      ;[collectedErrors, expectedErrors] = this._processTransactionExceptions(
+        transaction,
+        collectedErrors,
+        expectedErrors
+      )
+    } else if (isErroredTransaction) {
+      ;[collectedErrors, expectedErrors] = this._processTransactionErrors(
```

Comment:
What are these semicolons doing here? Did eslint put them there?

Reply:
since we do not use semi-colons, this is necessary for destructuring of let variables.  The AST parser does not know how to handle it without it

Reply:
ok, I don't think this is necessary. It's only needed when object destructuring

Reply:
I don't think this is an AST parser issue but how ESLint parses code without semi colons